corehq/apps/userreports/reports/filters/choice_providers.py

```diff
@@ -402,8 +402,11 @@ def default_value(self, user):
 class DomainChoiceProvider(ChainableChoiceProvider):
 
     @memoized
-    def _query_domains(self, domain, query_text):
+    def _query_domains(self, domain, query_text, user=None):
         domains = {domain}
+        if user:
+            if not user.can_view_registry_data(self.report.registry_helper.registry_slug):
```

Comment:
I don't think the user has a `can_view_registry_data` method (or dynamic dispatch which would make this work).
```suggestion
            if not RegistryPermissionCheck(domain, user).can_view_registry_data(self.report.registry_helper.registry_slug):
```

Note: import of `RegistryPermissionCheck` required in the module.